lemonspotter/core/function.py

```diff
@@ -41,78 +47,134 @@ def name(self) -> str:
         """This property provides access to the Function name."""
 
         if 'name' not in self._json:
-            raise Exception('Function name is not in JSON.')
+            raise RuntimeError('Function name is not in JSON.')
 
         return self._json['name']
 
     @property
     def has_parameters(self) -> bool:
         """"""
 
-        return self._json.get('parameters', False)
+        return 'parameters' in self._json
+
+    @property
+    def has_in_parameters(self) -> bool:
+        """"""
+
+        return len(self.in_parameters) > 0
+
+    @property
+    def has_inout_parameters(self) -> bool:
+        """"""
+
+        return len(self.inout_parameters) > 0
+
+    @property
+    def has_out_parameters(self) -> bool:
+        """"""
+
+        return len(self.out_parameters) > 0
 
     @property  # type: ignore
-    @lru_cache()
     def parameters(self) -> Sequence[Parameter]:
         """This property provides access to the parameter list of this Function object."""
 
         if 'parameters' not in self._json:
-            raise Exception('Parameters are not in JSON.')
+            raise RuntimeError('Parameters are not in JSON.')
+
+        if self._parameters is None:
+            self._parameters = tuple(Parameter(parameter)
+                                     for parameter in self._json['parameters'])
+
+        return self._parameters
+
+    @property
+    def in_parameters(self) -> Sequence[Parameter]:
+        """
+        Find all parameters which are with IN or INOUT parameters.
+        """
+
+        if self._in_parameters is None:
+            self._in_parameters = tuple(parameter
+                                        for parameter in self.parameters
+                                        if parameter.direction is Direction.IN)
+
+        return self._in_parameters
+
+    @property
+    def inout_parameters(self) -> Sequence[Parameter]:
+        """
+        Find all parameters which are with IN or INOUT parameters.
```

Comment:
Shouldn't this be "Find all parameters which are with INOUT parameters"

Reply:
Yes, I originally had them merged, but then decided to separate them.